### PR TITLE
 pre_defined_classes-simple_programs_taking_numeric_user_input-medium-q10

### PR DESCRIPTION
**Pull request by yuqi88**
## PR Checklist
- [ ]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [ ]  Are the assertion fail messages good?
    - [ ]  Logical (does it make sense?)
    - [ ]  Readable (can a student understand what’s wrong?)
    - [ ]  Not too feed-y (does it give away the answer?)
    - [ ]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [ ]  Specific (does the fail message relate to the question?)
- [ ]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [ ]  EOF newlines
- [ ]  Do the comments for Parsons follow the correct convention?
- [ ]  Is the question being tested sufficiently?
    - [ ]  Are there representative cases to test the program?
    - [ ]  What about edge cases? (empty strings, negatives, infinity, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
---

## Not yet, but soon

- [ ]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”

### CODE REVIEW

**Comment by yuqi88** [comment]
The wording regarding the gas part is very confusing. 
If the weight of the gas we are asking for is just one-way trip and not around-trip, why do we need to /2 the gas?

File: src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+public class PiperSaratoga {
+    public static void main( String[] argos )
+    {
+        final int MAX_CAPACITY = 1200;
+        Scanner scan = new Scanner( System.in );
+        System.out.println( "How much do all the travellers weigh?" );
+        double people = scan.nextDouble();
+        System.out.println( "How much does the gas weigh for the return flight?" );
+        double gas = scan.nextDouble();
+        gas = gas / 2;
```
Comment (Hedgemon4) [inline src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java:49]: Yeah, you can just take out the gas / 2, it does not make sense

File: src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package pre_defined_classes.simple_programs_taking_numeric_user_input.medium.q10;
+
+/*
+A Piper Saratoga is a hobby plane with six seats that has a capacity to hold 1200 pounds on a flight.
+ Make a program that helps this plane's pilot determine how much luggage can be brought on the return flight from
```
Comment (Hedgemon4) [inline src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java:5]: Also, the question asks how much luggage they can take back, not souvenirs, so I would just prompt the user for the weight of the gas and people, and then print out how heavy their luggage can be.

File: src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    //huh?
+    How much do all the travellers weigh?
+    510
+    How much does the gas weigh for the return flight?
+    800
+    What is the total weight of luggage from all travelers?
+    300
+    You can bring back -10.0 pounds of souvenirs!
+
+    //huh?
```
Comment (Hedgemon4) [inline src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java:28]: Is the `// huh?` for the negative numbers?

File: src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    //Parson
```
Comment (Hedgemon4) [inline src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/MainTest.java:18]: `// Parsons`

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java (hunk 3 above)
Comment (yuqi88) [reply src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java:28]: My bad. the second `huh?` was a wording problem in the question that I've fixed. Forgot to delete my notes.

File: src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/MainTest.java (hunk 4 above)
Comment (Hedgemon4) [reply src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/MainTest.java:19]: The label is still incorrect in the file

File: src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+Assuming users only enter valid inputs, meaning that the weight of the travellers must be >= 0 and that gas must be > 0.
+
+Sample output:
+    How much do all the travellers weigh in pounds?
+    376
+    How much does the gas weigh for the return flight in pounds?
+    700
+    You can board with and check in 124.0 pounds of luggage in total.
+
+    How much do all the travellers weigh in pounds?
```
Comment (Hedgemon4) [inline src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java:17]: Only need one sample output

File: src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/MainTest.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        return Stream.of(Arguments.of(510, 800, -110.0), Arguments.of(376, 700, 124.0),  Arguments.of(210, 100, 890.0),
+                Arguments.of(0, 900, 300.0)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void calculateAvailableSouvenirsPounds(double people, double gas, double luggage ) throws InvalidClauseException {
+        runWithInput(people + " "+ gas);
+        assertEquals(Double.parseDouble(getItemByName("luggage")), luggage, "Incorrect printing out the allowed pounds of luggage.");
```
Comment (Hedgemon4) [inline src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/MainTest.java:46]: This error message does not make grammatical sense.

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/MainTest.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    // Parson
```
Comment (Hedgemon4) [inline src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/MainTest.java:19]: `// Parsons`

File: src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
+package pre_defined_classes.simple_programs_taking_numeric_user_input.medium.q10;
+
+/*
+A Piper Saratoga is a hobby plane with six seats that has a capacity to hold 1200 pounds on a flight.
+Make a program that helps this plane's pilot determine how much luggage can be brought on the return flight from
+a vacation. The program needs to consider the weight of the travellers and the weight of the gas needed for the return
+flight to determine the total weight of the luggage from all travelers for the return flight.
```
Comment (Hedgemon4) [inline src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java:7]: `to determine the total weight allowance for luggage on the return flight.`

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
+package pre_defined_classes.simple_programs_taking_numeric_user_input.medium.q10;
+
+/*
+A Piper Saratoga is a hobby plane with six seats that has a capacity to hold 1200 pounds on a flight.
+Make a program that helps this plane's pilot determine how much luggage can be brought on the return flight from
+a vacation. The program needs to consider the weight of the travellers and the weight of the gas needed for the return
+flight to determine the total weight allowance for luggage on the return flight.
+Assuming users only enter valid inputs, meaning that the weight of the travellers must be >= 0 and that gas must be > 0.
```
Comment (Hedgemon4) [inline src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/PiperSaratoga.java:8]:
```suggestion
Assume the user only enters valid inputs, meaning that the weight of the travellers must be >= 0 and that gas must be > 0.
```

File: src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/MainTest.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(Arguments.of(510, 800, -110.0), Arguments.of(376, 700, 124.0),  Arguments.of(210, 100, 890.0),
+                Arguments.of(0, 900, 300.0)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void calculateAvailableLuggagePounds(double people, double gas, double luggage ) throws InvalidClauseException {
```
Comment (Hedgemon4) [inline src/pre_defined_classes/simple_programs_taking_numeric_user_input/medium/q10/MainTest.java:44]:
```suggestion
    public void calculatesAvailableLuggageWeightCorrectly(double people, double gas, double luggage ) throws InvalidClauseException {
```

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — APPROVED** [review]

**Pull request merged**